students/kuhnbt/lesson8/circle_class.py

```diff
@@ -0,0 +1,93 @@
+from math import pi
+from functools import total_ordering
+
+
+@total_ordering
+class Circle:
+    """A class representing a circle"""
+    def __init__(self, radius):
+        if radius <= 0:
+            raise ValueError('Radius must be positive')
+        self._radius = radius
+        self._diameter = self.radius * 2
+        self._area = None
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @radius.setter
+    def radius(self, value):
+        if value <= 0:
+            raise ValueError('Radius must be positive')
+        self._radius = value
+        self._diameter = value * 2
+
+    @property
+    def diameter(self):
+        return self._diameter
+
+    @diameter.setter
+    def diameter(self, value):
+        if value <= 0:
+            raise ValueError('Diameter must be positive')
+        self._diameter = value
+        self._radius = value / 2
+
+    @property
+    def area(self):
+        return self._radius**2 * pi
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        if diameter <= 0:
+            raise ValueError('Diameter must be positive')
+        self = cls(diameter / 2)
+        self._diameter = diameter
+        return self
```

Comment:
it is typically just this -> `return cls(diameter / 2)`

diameter is already set in the `__init__` function so need need to set it again